maven-plugin/src/main/java/org/openmrs/maven/plugins/AddDependency.java

```diff
@@ -0,0 +1,134 @@
+package org.openmrs.maven.plugins;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.openmrs.maven.plugins.model.DistroProperties;
+
+import org.openmrs.maven.plugins.utility.NPMHelper;
+import org.openmrs.maven.plugins.utility.SDKConstants;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+@Mojo(name = "add-dependency", requiresProject = false)
+public class AddDependency extends AbstractTask {
+
+    /**
+     *  Path to the openmrs-distro.properties file to modify
+     */
+    @Parameter(property = "distro")
+    private String distro;
+
+    /**
+     * Type of the dependency to add
+     */
+    @Parameter(property = "type")
+    private String type;
+
+    /**
+     * Maven group id of the dependency
+     */
+    @Parameter(property = "groupId")
+    private String groupId;
+
+    /**
+     * Maven artifact id of the dependency
+     */
+    @Parameter(property = "artifactId")
+    private String artifactId;
+
+    /**
+     * Version of the dependency
+     */
+    @Parameter(property = "version")
+    private String version;
+
+    /**
+     * Name of the frontend module
+     */
+    @Parameter(property = "moduleName")
+    private String moduleName;
+
+
+    private final String DEPENDENCY_TYPE_PROMPT = "Enter the type of dependency you need to add";
+
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException  {
+        if (distro == null) {
+            File userDir = new File(System.getProperty("user.dir"));
+            File distroFile = new File(userDir, DistroProperties.DISTRO_FILE_NAME);
+            if (distroFile.exists()) {
+                distro = distroFile.getAbsolutePath();
+            }
+        }
+
+        if(StringUtils.isBlank(type)) {
```

Comment:
Shouldn't we also prompt for this in the case that the `type` doesn't match one of the known types?